soc/xtensa/intel_adsp/common/include/soc.h

```diff
@@ -112,4 +119,15 @@ extern bool soc_cpus_active[CONFIG_MP_NUM_CPUS];
  */
 int soc_adsp_halt_cpu(int id);
 
+static inline bool intel_adsp_ptr_executable(const void *p)
+{
+	return p >= (void *)_text_start && p <= (void *)_text_end;
```

Comment:
There is some code in IMR too, my guess this is going to e.g. suppress backtraces of the boot code?  The architecure doesn't have "executable" mappings per se, any memory might be executable and contain symbols.

Reply:
For IMR, addressed on v2. Forced a panic on boot and saw some backtrace.